### PR TITLE
StatefulSet Integration Test

### PR DESCRIPTION
**Pull request by ryanemerson**
https://issues.redhat.com/browse/ISPN-11527
https://issues.redhat.com/browse/ISPN-11537
https://issues.redhat.com/browse/ISPN-11538
https://issues.redhat.com/browse/ISPN-11582

I'll create a backport PR once this has been merged.

The clustered locks changed was required as previously it was not cleaning up it's resource and was blocking shutdown via sigterm.

ISPN-11538 should probably be 11.x only.

### CODE REVIEW

File: server/testdriver/core/src/main/java/org/infinispan/server/test/core/ContainerInfinispanServerDriver.java [hunk 1]
```diff
@@ -318,6 +329,16 @@ public void kill(int server) {
       Exceptions.unchecked(() -> containers.get(server).execInContainer(INFINISPAN_SERVER_HOME + "/bin/kill.sh"));
    }
 
+   // TODO should this just replace stop?
+   public void sigterm(int server) {
+      GenericContainer container = containers.get(server);
+      CountdownLatchLoggingConsumer latch = new CountdownLatchLoggingConsumer(1, SHUTDOWN_MESSAGE_REGEX);
+      container.withLogConsumer(latch);
+      Container.ExecResult result = Exceptions.unchecked(() -> container.execInContainer(INFINISPAN_SERVER_HOME + "/bin/term.sh"));
+      System.out.printf("[%d] TERM %s\n", server, result);
```
Comment (karesti) [inline server/testdriver/core/src/main/java/org/infinispan/server/test/core/ContainerInfinispanServerDriver.java]: we want this ?

**Comment by karesti** [comment]
@ryanemerson I reviewed all, LGTM maybe someone else should have a look to the server part if you need to. If CI is good, it's fine from my side, just a small question

File: server/testdriver/core/src/main/java/org/infinispan/server/test/core/ContainerInfinispanServerDriver.java (hunk 1 above)
Comment (ryanemerson) [reply server/testdriver/core/src/main/java/org/infinispan/server/test/core/ContainerInfinispanServerDriver.java]: It's what we do in the pause and resume methods. Adding a log statement doesn't result in any output in the test logs for some reason.

**Comment by karesti** [comment]
StatefulSetRollingUpgradeTest is failing @ryanemerson

**Comment by ryanemerson** [comment]
> StatefulSetRollingUpgradeTest is failing @ryanemerson

Yeah that's failing on master. I need to look into it.

**Comment by karesti** [comment]
ok, so I merge then

**Comment by tristantarrant** [comment]
You want the pgrep fixes also for pause.sh and resume.sh

**Comment by ryanemerson** [comment]
@tristantarrant updated

File: server/runtime/src/main/server/bin/server.sh [hunk 2]
```diff
@@ -11,52 +11,43 @@ DIRNAME=`dirname "$0"`
...
-      trap "kill -HUP  $ISPN_PID" HUP
-      trap "kill -TERM $ISPN_PID" INT
-      trap "kill -QUIT $ISPN_PID" QUIT
-      trap "kill -PIPE $ISPN_PID" PIPE
-      trap "kill -TERM $ISPN_PID" TERM
-      if [ "x$ISPN_PIDFILE" != "x" ]; then
-        echo $ISPN_PID > $ISPN_PIDFILE
+   # Execute the JVM in the background
+   eval \"$JAVA\" $JAVA_OPTS \
+      -Djava.util.logging.manager=org.apache.logging.log4j.jul.LogManager \
```
Comment (tristantarrant) [inline server/runtime/src/main/server/bin/server.sh:16]: You have lost the `visualvm.display.name`

Comment (ryanemerson) [reply server/runtime/src/main/server/bin/server.sh:17]: It seems it was never set when launching in the background. Updated :+1:

**Pull request merged**